scripts/manila-consistency.py

```diff
@@ -137,6 +137,31 @@ def fix_wrong_share_gtstm(meta, wrong_share_gtstm):
             values(updated_at=now, deleted_at=now, deleted=share_gtstm_id)
         delete_share_gtstm_q.execute()
 
+# get all the rows with a share_instance_access_map still defined where the corresponding share_instance is already deleted
+def get_wrong_share_instance_access_mapping(meta):
+    wrong_share_instance_access_mapping = {}
+    share_instance_access_mapping_t = Table('share_instance_access_map', meta, autoload=True)
+    share_instances_t = Table('share_instances', meta, autoload=True)
+    share_instance_access_mapping_join = share_instance_access_mapping_t.join(share_instances_t,share_instance_access_mapping_t.c.share_instance_id == share_instances_t.c.id)
+    wrong_share_instance_access_mapping_q = select(columns=[share_instances_t.c.id,share_instances_t.c.deleted,share_instance_access_mapping_t.c.id,share_instance_access_mapping_t.c.deleted]).select_from(share_instance_access_mapping_join).where(and_(share_instances_t.c.deleted != "False",share_instance_access_mapping_t.c.deleted == "False"))
+
+    # return a dict indexed by share_instance_access_mapping id and with the value share_instance_id for non deleted mappings
+    for (share_instance_id, share_instance_deleted, share_instance_access_mapping_id, share_instance_access_mapping_deleted) in wrong_share_instance_access_mapping_q.execute():
```

Comment:
the unused variables could be removed? `(share_instance_id, _, share_instance_access_mapping_id, _) `